airbyte-server/src/main/java/io/airbyte/server/ServerApp.java

```diff
@@ -176,6 +179,18 @@ public void configure() {
     server.join();
   }
 
+  private static void createDeploymentIfNoneExists(final JobPersistence jobPersistence) throws IOException {
```

Comment:
to make sure I understand: it makes sense to have the server do this because the scheduler's start up depends on the server. this way, we know the scheduler can access the deployment id as soon as it starts up?

Reply:
yeah. at least how we have everything structured now, we know that the main method of the server runs first and scheduler waits for the database to be in a ready state (in other words fully set up by the sever) before it does anything.